### PR TITLE
Branch delete feature

### PR DESCRIPTION
**Pull request by npazosmendez**
@danifujii

### CODE REVIEW

File: branchsearch.cpp [hunk 1]
```diff
@@ -234,8 +249,12 @@ int main(int argc, char** argv){
                 if(selected_branch > 0) selected_branch--;
             }else if (c == KEY_DOWN){
                 if(selected_branch < filtered_branches.size()-1 && selected_branch < MAX_BRANCH_SHOWN - 1) selected_branch++;
+            }else if (c == KEY_DC){
```
Comment (danifujii) [inline branchsearch.cpp:255]: Supporting backspace can't hurt

File: branchsearch.cpp [hunk 2]
```diff
@@ -109,6 +110,19 @@ void print_window(char* regex_value, vector<branch_t*> filtered_branches, int se
...
 }
 
+void delete_branch(vector<branch_t*> &branches, int target_index){
+    erase();
+    string branch_name = branches[target_index]->name;
+    printw("Delete '%s'? [Y/N]\n", branch_name.c_str());
+    while (1) {
+        if (kbhit()) {
+            int c = getch();
+            if(c == 'Y' or c == 'y') run_command("git branch -d " + branch_name + " 2>&1");
```
Comment (danifujii) [inline branchsearch.cpp:123]: The fact that it takes you out of `bs` when this fails is pretty jarring TBH.

File: branchsearch.cpp (hunk 1 above)
Comment (npazosmendez) [reply branchsearch.cpp:255]: Backspace is reserved for typing

Comment (danifujii) [reply branchsearch.cpp:255]: But couldn't it be used if nothing was typed so far? I thought about using it but your point is fair.

Comment (npazosmendez) [reply branchsearch.cpp:255]: Mmm I think that’s kind of arbitrary and confusing

**Review by danifujii — APPROVED** [review]

**Pull request merged**